sql: allow extra columns to be specified for COPY

otan opened the pull request:
The intention is for this to be used with AWS DMS in the PGURL with
the additional `--options=-c%20copy_num_trailing_fields=1`
speciifed.

Refs: #78688

Release note (sql change): Introduce a `copy_num_trailing_fields`
variable, which allows extra trailing columns for COPY.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/78688)
<!-- Reviewable:end -->

otan committed:
sql: allow extra columns to be specified for COPY

The intention is for this to be used with AWS DMS in the PGURL with
the additional `--options=-c%20copy_num_trailing_fields=1`
speciifed.

Release note (sql change): Introduce a `copy_num_trailing_fields`
variable, which allows extra trailing columns for COPY.
